### ISSUE TITLE
Timestamp as observation

### ISSUE THREAD

**Issue by AminDar** [Stable-Baselines-Team/stable-baselines3-contrib#186]
### ❓ Question

Since LSTM is used to recognize patterns in sequences of data, such as time series ( let's say the temperature variation in a room over a year), I wonder why we can not pass a "Timestamp" to observations and env_checker gives an error:

TypeError: float() argument must be a string or a number, not 'Timestamp'

observation box:
` self.observation_space = spaces.Box( low=-np.inf, high=np.inf, shape=(6,) )`

My question is if we must change the timestamps to numerical values if we have such a time series:
```

    2021-03-02 00:00:00                           To                0             
    2021-03-02 01:00:00                           To                60 
    2021-03-02 02:00:00                           To                120 
    2021-03-02 03:00:00                           To                180
    2021-03-02 04:00:00                           To                240
    ....   
    ....
    2021-04-30 22:00:00                           To                1320
    2021-04-30 23:00:00                           To                1380 
```

Or we should use hot-encoder? and which one would make more sense  to track the temporal aspect of the temperature changes?

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/issues) in the repo
- [X] I have read the [documentation](https://sb3-contrib.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> TypeError: float() argument must be a string or a number, not 'Timestamp'
> Or we should use hot-encoder?

yes, you need to encode it, a `Timestamp` object is not a float or numpy array (that's what is expected when you define a `spaces.Box`).

**Comment by AminDar** [comment]
I wanted to share my experience with the [PPO with recurrent policy (RecurrentPPO aka PPO LSTM)](https://ppo-details.cleanrl.dev//2021/11/05/ppo-implementation-details/). 
I had a 72h hourly timestamp data which I encoded it to cyclical-features , (same method described [here](https://ianlondon.github.io/blog/encoding-cyclical-features-24hour-time/).

A simple description of the environment:
the agent has to learn to alter the set points for a thermostat in such a way that the temperature in the room falls within a range. The time of the day is crucial since the temperature outside varies during the day. 

there are other observations (6more) but I skip describing them. So I trained the agent once without giving the time features and once with.
I was expecting that providing the cyclical-features would improve the overall performance of the agent dramatically, but it didn't. I get almost the same results if I don't provide the cyclical features.

I wonder if it's a normal behavior or not.

p.s : The best result is achieved when training with SAC, which is not a LSTM, recurrent, policy!
